swagger2.0/oic.r.pushreceiver.swagger.json

```diff
@@ -0,0 +1,263 @@
+          "204": {
+            "description": "whole receiver objects are removed successfully\n"
+          }
+        }
+      }
+    },
+    "/PushReceiverResURI?receiveruri=<uri>": {
+      "post": {
+        "description": "Updates a push receiver object which has `receiveruri`\n",
+        "parameters": [
+          { "$ref": "#/parameters/interface-rw" },
+          { "$ref": "#/parameters/receiveruri" },
+          { "$ref": "#/parameters/body-receiver-update" }
+        ],
+        "responses": {
+          "204": {
+            "description": "a receiver object is updated successfully\n"
+          }
+        }
+      },
+      "delete": {
+        "description": "delete a push receiver object which has `receiveruri`\n",
+        "parameters": [
+          { "$ref": "#/parameters/interface-rw" },
+          { "$ref": "#/parameters/receiveruri" }
+        ],
+        "responses": {
+          "204": {
+            "description": "a receiver object was removed successfully\n"
+          }
+        }
+      }
+    },
+    "/PushReceiverResURI?if=oic.if.baseline": {
+      "get": {
+        "description": "Resource that defines the receiver for Push Notifications",
+        "parameters": [
+          { "$ref": "#/parameters/interface-baseline" }
+        ],
+        "responses": {
+          "200": {
+            "description": "",
+            "x-example": {
+              "rt": ["oic.r.pushreceiver"],
+              "if": ["oic.if.rw", "oic.if.baseline"],
+              "receivers": [
+                {
+                  "receiveruri": "/mylocaltargeturiforthermostats",
+                  "rts": ["oic.r.temperature", "oic.r.humidity"]
+                },
+                {
+                  "receiveruri": "/mylocaltargeturifordontcare",
+                  "rts": []
+                }
+              ]
+            },
+            "schema": { "$ref": "#/definitions/get-baseline-response" }
+          }
+        }
+      }
+    }
+  },
+  "parameters": {
+    "interface-baseline": {
+      "in": "query",
+      "name": "if",
+      "type": "string",
+      "enum": ["oic.if.baseline"]
+    },
+    "interface-rw": {
+      "in": "query",
+      "name": "if",
+      "type": "string",
+      "enum": ["oic.if.rw"]
+    },
+    "receiveruri": {
+      "in": "query",
+      "name": "receiveruri",
+      "type": "string"
+    },
+    "body-receiver-update": {
+      "in": "body",
+      "name": "receiver",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/receiver"
+      },
+      "x-example": {
+        "receiveruri": "/mylocaltargeturifordontcare", 
+        "rts": []
+      }
+    },
+    "body-receivers-update": {
+      "in": "body",
+      "name": "receivers",
+      "required": true,
+      "schema": {
+        "$ref": "#/definitions/post-rw-request"
+      },
+      "x-example": {
+        "receivers": [
+          { 
+            "receiveruri": "/mylocaltargeturifordontcare", 
+            "rts": []
```

Comment:
should there be a value here (as example)?

Reply:
do you mean `"rts"`? 
nop, `"rts"` can bi empty (`"minItems": 0`). (see "#/definitions/receivers")

If `"rts"` is empty, any type of Resource can be posted to the corresponding `"receiveruti"`